Library/Homebrew/os/mac.rb

```diff
@@ -195,10 +195,15 @@ def self.macports_or_fink
 
     sig { params(ids: String).returns(T.nilable(Pathname)) }
     def self.app_with_bundle_id(*ids)
-      path = mdfind(*ids)
-             .reject { |p| p.include?("/Backups.backupdb/") }
-             .first
-      Pathname.new(path) if path.present?
+      require "bundle_version"
+
+      paths = mdfind(*ids).filter_map do |bundle_path|
+        Pathname.new(bundle_path) if bundle_path.exclude?("/Backups.backupdb/")
+      end
+      return paths.first if paths.any? { |bp| !(bp/"Contents/Info.plist").exist? }
```

Comment:
```suggestion
      return paths.first if paths.none? { |bp| (bp/"Contents/Info.plist").exist? }
```
maybe?

Reply:
I think that's not equivalent, unfortunately. In particular, suppose we have an `Foo.app/Contents/Info.plist` but no `Foo_debug.app/Contents/Info.plist`.

Before your suggestion, we would take the first `Foo*.app` that `mdfind` returns. After your suggestion, we would try to check the `Info.plist`s for the newest one, but fail (because `Foo_debug.app` has no `Info.plist`).

For a change that is equivalent, we can do:
```suggestion
      return paths.first unless paths.all? { |bp| (bp/"Contents/Info.plist").exist? }
```

Reply:
Yeh, suggested work makes sense!